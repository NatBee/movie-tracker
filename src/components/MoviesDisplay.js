import React from 'react';
import { Link } from 'react-router-dom';
import Movie from './Movie';


const MoviesDisplay = () => {

  return (
    <div>
      <p>I'm the all the movies!</p>
      <Movie />
    </div>
  )
}

export default MoviesDisplay;