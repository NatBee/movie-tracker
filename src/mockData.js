export const mockData = [{
  "vote_count": 20,
  "id": 337167,
  "video": false,
  "vote_average": 6.4,
  "title": "Fifty Shades Freed",
  "popularity": 45.670468,
  "poster_path": "/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg",
  "original_language": "en",
  "original_title": "Fifty Shades Freed",
  "genre_ids": [18, 10749],
  "backdrop_path": "/iuyE1qy9HoTsMg0DHB1KJxM94DJ.jpg",
  "adult": false,
  "overview": "The third installment of the Fifty Shades Trilogy.",
  "release_date": "2018-02-07"
}];