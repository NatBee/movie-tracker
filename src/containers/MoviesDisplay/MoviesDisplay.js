import React, { Component } from 'react';
import { loadMovies, toggleFavorite, showFavorites } from '../../actions/index.js';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { getMovies } from '../../helper/apiCall';
import Movie from '../Movie/Movie';
import './MoviesDisplay.css'

export class MoviesDisplay extends Component {
  async componentDidMount() {
    const results = await getMovies();
    this.props.loadMovies(results);
  }

  renderedMovies = () => this.props.movies.map( ( movie ) => {
    const rating = Math.round(movie.popularity);

    return (
      <Movie key={ movie.id }
             poster={ movie.poster_path }
             title={ movie.title }
             rating={ rating }
             overview={ movie.overview }
             releaseDate={ movie.release_date }
             id={ movie.id } />
    )
  })

  render() {
    return (
      <div className="container">
        { this.renderedMovies() }
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  movies: store.movies
})

export const mapDispatchToProps = (dispatch) => ({
  loadMovies: movies => dispatch(loadMovies(movies)),
  showFavorites: filter => dispatch(showFavorites(filter))
})

export default connect(mapStateToProps, mapDispatchToProps)(MoviesDisplay);