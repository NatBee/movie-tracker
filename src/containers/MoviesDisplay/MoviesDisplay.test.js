/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import { MoviesDisplay, mapStateToProps, mapDispatchToProps } from './MoviesDisplay';
import { mockData } from '../../mockData'

describe('MoviesDisplay', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MoviesDisplay movies={ mockData }/>);
  });

  it('should match snapshot',() => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map to the store correctly', () => {
    const mockStore = {
      movies: mockData
    }
    const mapped = mapStateToProps(mockStore)

    expect(mapped).toEqual(mockStore)
  });

  it('should call the dispatch function when using a function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.loadMovies();
    mapped.showFavorites();

    expect(mockDispatch).toHaveBeenCalled();
  });
})