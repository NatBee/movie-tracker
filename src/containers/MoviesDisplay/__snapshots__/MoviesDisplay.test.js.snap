// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesDisplay should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesDisplay
    movies={
      Array [
        Object {
          "adult": false,
          "backdrop_path": "/iuyE1qy9HoTsMg0DHB1KJxM94DJ.jpg",
          "favorite": false,
          "genre_ids": Array [
            18,
            10749,
          ],
          "id": 337167,
          "original_language": "en",
          "original_title": "Fifty Shades Freed",
          "overview": "The third installment of the Fifty Shades Trilogy.",
          "popularity": 45.670468,
          "poster_path": "/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg",
          "release_date": "2018-02-07",
          "title": "Fifty Shades Freed",
          "video": false,
          "vote_average": 6.4,
          "vote_count": 20,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Movie))
          id={337167}
          overview="The third installment of the Fifty Shades Trilogy."
          poster="/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg"
          rating={46}
          releaseDate="2018-02-07"
          title="Fifty Shades Freed"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "337167",
        "nodeType": "function",
        "props": Object {
          "id": 337167,
          "overview": "The third installment of the Fifty Shades Trilogy.",
          "poster": "/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg",
          "rating": 46,
          "releaseDate": "2018-02-07",
          "title": "Fifty Shades Freed",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Movie))
            id={337167}
            overview="The third installment of the Fifty Shades Trilogy."
            poster="/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg"
            rating={46}
            releaseDate="2018-02-07"
            title="Fifty Shades Freed"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "337167",
          "nodeType": "function",
          "props": Object {
            "id": 337167,
            "overview": "The third installment of the Fifty Shades Trilogy.",
            "poster": "/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg",
            "rating": 46,
            "releaseDate": "2018-02-07",
            "title": "Fifty Shades Freed",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
